src/v0/destinations/impact_radius/transform.js

```diff
@@ -0,0 +1,285 @@
+const get = require("get-value");
+const sha1 = require("js-sha1");
+const btoa = require("btoa");
+const { EventType } = require("../../../constants");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const {
+  checkConfigurationError,
+  populateProductProperties,
+  populateAdditionalParameters
+} = require("./util");
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  simpleProcessRouterDest,
+  isAppleFamily,
+  constructPayload,
+  isDefinedAndNotNull,
+  removeUndefinedAndNullValues,
+  isDefinedAndNotNullAndNotEmpty
+} = require("../../util");
+const {
+  ConfigurationError,
+  TransformationError,
+  InstrumentationError
+} = require("../../util/errorTypes");
+
+const responseBuilder = (payload, endpoint, Config) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = endpoint;
+    response.headers = {
+      "Content-Type": "application/x-www-form-urlencoded",
+      Authorization: `Basic ${btoa(`${Config.accountSID}:${Config.apiKey}`)}`
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.FORM = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+  throw new TransformationError(
+    "Payload could not be populated due to wrong input"
+  );
+};
+
+const buildPageLoadResponse = (
+  message,
+  campaignId,
+  impactAppId,
+  enableEmailHashing
+) => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.PAGELOAD.name]
+  );
+  if (isDefinedAndNotNull(payload.CustomerEmail)) {
+    payload.CustomerEmail = enableEmailHashing
+      ? sha1(payload?.CustomerEmail)
+      : payload?.CustomerEmail;
+  }
+  payload.CampaignId = campaignId;
+  if (isDefinedAndNotNullAndNotEmpty(impactAppId)) {
+    payload.PropertyId = impactAppId;
+    payload.ImpactAppId = impactAppId;
+  }
+
+  const os = get(message, "context.os.name");
+  if (os && isAppleFamily(os.toLowerCase())) {
+    payload.AppleIfv = get(message, "context.device.id");
+    payload.AppleIfa = get(message, "context.device.advertisingId");
+  } else if (os && os.toLowerCase() === "android") {
+    payload.AndroidId = get(message, "context.device.id");
+    payload.GoogAId = get(message, "context.device.advertisingId");
+  }
```

Comment:
this part of the code is repeating below. create a util function and reuse at both places